### ISSUE TITLE
Bad error message when not logged in to container registry

### ISSUE THREAD

**Issue by rynowak** [radius-project/radius#6297]
### Steps to reproduce (required)

- Let your ACR login expire
- Run the following command

```sh
rad bicep publish --file test/functional/corerp/resources/testdata/recipes/test-recipes/dapr-state-store.bicep --target br:rynowak.azurecr.io/test/functional/corerp/recipes/dapr-state-store:latest
```

Any source and destination are fine, as long as you are not logged in to the destination.

### Observed behavior (required)

```sh
Building test/functional/corerp/resources/testdata/recipes/test-recipes/dapr-state-store.bicep...
Error: Failed to publish Bicep file: HEAD "https://rynowak.azurecr.io/v2/test/functional/corerp/recipes/dapr-state-store/manifests/sha256:10131c538619da7466e8ee9fbf573295cecfab81cd46b544bfbe058cd3347cee": POST "https://rynowak.azurecr.io/oauth2/token": response status code 401: Unauthorized
```

### Desired behavior (required)

A better error message, that explains the problem and tells you how to log in. If I were an external customer I wouldn't be sure how to fix this.

[AB#7968](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/7968)

[AB#9502](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/9502)

**Comment by lechnerc77** [comment]
Is this already fixed? I just retried it and the result is: 

```bash
rad bicep publish --file test/functional/samples/testdata/tutorial-environment.bicep --target br:rynowak.azurecr.io/test/functional/corerp/recipes/dapr-state-store:latest
Building test/functional/samples/testdata/tutorial-environment.bicep...
Failed to publish Bicep file "test/functional/samples/testdata/tutorial-environment.bicep" to "rynowak.azurecr.io/test/functional/corerp/recipes/dapr-state-store:latest". Cause: POST "https://rynowak.azurecr.io/v2/test/functional/corerp/recipes/dapr-state-store/blobs/uploads/": response status code 401: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information.: [map[Action:pull Name:test/functional/corerp/recipes/dapr-state-store Type:repository] map[Action:push Name:test/functional/corerp/recipes/dapr-state-store Type:repository]].
```

Still quite some text, but the reference to `https://aka.ms/acr/authorization` is helpful, so wondering if the error message should be further improved?

**Comment by rynowak** [comment]
The error message coming from ACR does include a link to the instructions, but most users don't notice when they hit this.

We had a _similar_ issue reported here: https://github.com/radius-project/radius/issues/6902 which is evidence that this is still confusing. 

---

If you look at the code, we're not doing anything special to handle failed logins: https://github.com/radius-project/radius/blob/main/pkg/cli/cmd/bicep/publish/publish.go#L300

We're just trying to publish and then printing whatever error comes back from the server.

---

I think a good fix to this would attempt a login up front, and then "pretty print" the error that comes out using `clierrors.MessageWithCause`: https://github.com/radius-project/radius/blob/main/pkg/cli/clierrors/errors.go#L33

The error we print to the console should make a few things clear:

- Logging into your container registry failed (and which registry).
- You need to find instructions for your container registry and follow them.

Since users can use `rad bicep publish` with any OCI registry we can't build in knowledge of how login works to all of them. The users needs to do this separately from the `rad` CLI.

I think it some of these cases users aren't aware of what's happening behind the scenes (pulling/pushing from an OCI registry) so the error message is confusing.

**Comment by ytimocin** [comment]
Sounds good to me. The result of the call to `target.Push` should be "prettified" in a way that the Radius user can understand. cc/ @rynowak

**Comment by rynowak** [comment]
> Sounds good to me. The result of the call to `target.Push` should be "prettified" in a way that the Radius user can understand. cc/ @rynowak

I think that's a good option too. I was thinking we'd explicitly try to auth (new code) so we know whether it not it succeeds easily.

**Comment by lechnerc77** [comment]
Thanks for the input. If that's okay, I would try to provide a fix for that.

**Comment by gpltaylor** [comment]
> Is this already fixed? I just retried it and the result is:
> 
> ```shell
> Building test/functional/samples/testdata/tutorial-environment.bicep...
> Failed to publish Bicep file "test/functional/samples/testdata/tutorial-environment.bicep" to "rynowak.azurecr.io/test/functional/corerp/recipes/dapr-state-store:latest". Cause: POST "https://rynowak.azurecr.io/v2/test/functional/corerp/recipes/dapr-state-store/blobs/uploads/": response status code 401: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information.: [map[Action:pull Name:test/functional/corerp/recipes/dapr-state-store Type:repository] map[Action:push Name:test/functional/corerp/recipes/dapr-state-store Type:repository]].
> ```
> 
> Still quite some text, but the reference to `https://aka.ms/acr/authorization` is helpful, so wondering if the error message should be further improved?

This URL is useless. It doesn't document how to login to the ACR.

**Comment by rynowak** [comment]
> >   
> > Still quite some text, but the reference to `https://aka.ms/acr/authorization` is helpful, so wondering if the error message should be further improved?
> 
> This URL is useless. It doesn't document how to login to the ACR.

Good to know, I didn't click it 😆 

@lechnerc77 - looks like @gpltaylor already started working on a fix for this here: https://github.com/radius-project/radius/pull/6974#event-11328467395

**Comment by rynowak** [comment]
Fixed by #6974

**Comment by rynowak** [comment]
Thanks @gpltaylor !
